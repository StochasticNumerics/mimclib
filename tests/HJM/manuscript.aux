\relax 
\providecommand\hyper@newdestlabel[2]{}
\providecommand\HyperFirstAtBeginDocument{\AtBeginDocument}
\HyperFirstAtBeginDocument{\ifx\hyper@anchor\@undefined
\global\let\oldcontentsline\contentsline
\gdef\contentsline#1#2#3#4{\oldcontentsline{#1}{#2}{#3}}
\global\let\oldnewlabel\newlabel
\gdef\newlabel#1#2{\newlabelxx{#1}#2}
\gdef\newlabelxx#1#2#3#4#5#6{\oldnewlabel{#1}{{#2}{#3}}}
\AtEndDocument{\ifx\hyper@anchor\@undefined
\let\contentsline\oldcontentsline
\let\newlabel\oldnewlabel
\fi}
\fi}
\global\let\hyper@last\relax 
\gdef\HyperFirstAtBeginDocument#1{#1}
\providecommand\HyField@AuxAddToFields[1]{}
\providecommand\HyField@AuxAddToCoFields[2]{}
\citation{bjork2013monte}
\@writefile{toc}{\contentsline {section}{\tocsection {}{1}{Intro}}{1}{section.1}}
\citation{bjork2013monte}
\@writefile{toc}{\contentsline {section}{\tocsection {}{2}{Numerical tests}}{2}{section.2}}
\@writefile{lof}{\contentsline {figure}{\numberline {1}{\ignorespaces Sample realisation of a forward curve with the discretisation level $\overline  \ell = [5,5,5]^T$.\relax }}{3}{figure.caption.1}}
\bibstyle{agsm}
\bibdata{references}
\newlabel{tocindent-1}{0pt}
\newlabel{tocindent0}{0pt}
\newlabel{tocindent1}{19.4667pt}
\newlabel{tocindent2}{32.18062pt}
\newlabel{tocindent3}{0pt}
\providecommand*\caption@xref[2]{\@setref\relax\@undefined{#1}}
\newlabel{fig:gull}{{\caption@xref {fig:gull}{ on input line 176}}{4}{Numerical tests}{figure.caption.2}{}}
\newlabel{sub@fig:gull}{{}{4}{Numerical tests}{figure.caption.2}{}}
\newlabel{fig:tiger}{{\caption@xref {fig:tiger}{ on input line 183}}{4}{Numerical tests}{figure.caption.2}{}}
\newlabel{sub@fig:tiger}{{}{4}{Numerical tests}{figure.caption.2}{}}
\newlabel{fig:mouse}{{\caption@xref {fig:mouse}{ on input line 191}}{4}{Numerical tests}{figure.caption.2}{}}
\newlabel{sub@fig:mouse}{{}{4}{Numerical tests}{figure.caption.2}{}}
\newlabel{fig:mouse}{{\caption@xref {fig:mouse}{ on input line 196}}{4}{Numerical tests}{figure.caption.2}{}}
\newlabel{sub@fig:mouse}{{}{4}{Numerical tests}{figure.caption.2}{}}
\@writefile{lof}{\contentsline {figure}{\numberline {2}{\ignorespaces Empirical strong and weak convergence rate for the $\Delta t$ and $\Delta \tau $ discretisation errors.\relax }}{4}{figure.caption.2}}
\newlabel{fig:empInfDimRates}{{2}{4}{Empirical strong and weak convergence rate for the $\Delta t$ and $\Delta \tau $ discretisation errors.\relax }{figure.caption.2}{}}
