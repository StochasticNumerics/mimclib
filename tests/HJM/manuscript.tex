\documentclass[11pt]{amsart}
\usepackage{geometry}                % See geometry.pdf to learn the layout options. There are lots.
\geometry{a4paper}                   % ... or a4paper or a5paper or ... 
%\geometry{landscape}                % Activate for for rotated page geometry
%\usepackage[parfill]{parskip}    % Activate to begin paragraphs with an empty line rather than an indent
\usepackage{graphicx}
\usepackage{amssymb, amsmath, amsthm}
\usepackage{natbib}
\usepackage{epstopdf}
\usepackage{enumerate}
\usepackage{textcomp}
\usepackage{mathrsfs }
\usepackage[section]{algorithm}
\usepackage{algorithmic}
\usepackage{tikz}
\usepackage{pgfplots}
\usepgfplotslibrary{patchplots}
%\usepackage{subfigure}
\usepackage{subcaption}

\usepackage{hyperref}
\usepackage{comment}
\usepackage{multirow}

%line suggested by my compiler (Fabian)
\pgfplotsset{compat=1.9}

% put figures at the end of the document, while I am writing is very annoying to have the figures in the middle of the text! 
% \usepackage[nomarkers,figuresonly]{endfloat}


\DeclareGraphicsRule{.tif}{png}{.png}{`convert #1 `dirname #1`/`basename #1 .tif`.png}

\input{macros.tex}

\title[MIMC and MLMC HJM]{Multi-Index and Multi-Level Monte Carlo Evaluation of HJM Models}
\author{Juho H\"app\"ol\"a}
\address{\kaustaddress}
\email{juho.happola@iki.fi}


\date{\today}    
                                    % Activate to display a given date or no date
\begin{document}


\begin{abstract}
Notes of MIMC and MLMC evaluation of HJM models.
\end{abstract}

\maketitle
%\tableofcontents

\section{Intro}

We are interested in analysing the Forward Euler Finite Difference evaluation of forward curve models of HJM models. The main approach was represented with a convergence analysis in \cite{bjork2013monte}, whose notation we largely follow.

We are focused on the evolution of the forward curve $f \parent{t , \tau}$ for $t \in [0,t_{max}]$, $\tau \in [0,\tau_{max}]$ with $\tau_{max}>t_{max}$. In order to define a mesh for the numerical solution of the problem, let us define $\bar \ell \in \mathbb{N}^3$.

Let our quantity of interest be defined as the expectation of a functional $\mathcal F$
of the forward curve:
\begin{align*}
\mathcal F \parent f =& F \parent{\int_0 ^{t_{max}} f \parent{s,s} ds}
G \parent{ \int_{\tau_a}^{\tau_{max}} \Psi \parent{f \parent{t_{max},\tau}} d\tau }
\\
&+
\int_{0}^{t_{max}} F \parent{\int_0^{s} f \parent{s',s'} ds'} U \parent{f\parent{s,s}} ds.
\end{align*}

Let us first discretise the time interval as
\begin{align*}
\Delta t_{\bar \ell} &= 2^{-\ell_1} t_{max},
\\
t_{\bar \ell, n} &= n \Delta t_{\bar \ell}, ~~~ n \in [0,1,2,\cdots, 2^{\ell_1}].
\end{align*}
Since there is an additional requirement that $t_{max}$ be in
the mesh of maturities, one point in the mesh in $\tau \in [0,\tau_{max}]$. We let $\ell_1$ and $\ell_2$ define the number of discretisation points to the right and to the left of the fixed
point, respectively.
\begin{align*}
\Delta \tau_{\bar \ell,1} &= 2^{-\ell_3} t_{max},  \\
\Delta \tau_{\bar \ell,2} &= 2^{-\ell_2} \parent{\tau_{max}-t_{max}}, \\
\tau_{\bar \ell, n} &= n \Delta \tau t_{\bar \ell,1}, ~~~ n \in [0,1,2,\cdots, 2^{\ell_3}] \\
\tau_{\bar \ell, n} &= n \Delta \tau t_{\bar \ell,1}, ~~~ n \in [2^{\ell_3},2^{\ell_3}+1,\cdots, 2^{\ell_3}+2^{\ell 2}].
\end{align*}

Using this notation, we have the following rates of convergence, following the Multi-Index notation:
\begin{align}
\bar \beta &= [2,2,2]^T \\
\bar \gamma &= [1,1,1]^T .
\end{align} 
Choosing the Simpson method for integration, we have the order of quadrature $p_Q=3$, which brings the quadrature error to 
\begin{align}
\expp{\absval{\mathcal G\parent{\bar{\bar g}} - \mathcal G \parent{g}}^{2 \kappa}} \leq C \parent{\Delta \tau_{\bar \ell}}^{2 \kappa p_Q} = 
\tilde C 2^{-6 \kappa \mathrm{min} \parent{\ell_3,\ell_2}}.
\end{align}
On the other hand, the error of the simpson method is the sum of errors for $\tau < t_{max}$ and $\tau \leq t_{max}$. The error for these two domains are $\bigo{2^{-2 \ell_2}}$ and $\bigo {2^{-2 \ell_3}}$, respectively. Combining together, we get the weak and the strong rate
for the Simpson quadrature:
\begin{align}
\expp{\absval{\mathcal G\parent{\bar{\bar g}} - \mathcal G \parent{g}}^{2}} \leq&  \tilde C_Q 2^{-6 \mathrm{min} \parent{\ell_3,\ell_2}}
\\
\expp{\absval{\mathcal G\parent{\bar{\bar g}} - \mathcal G \parent{g}}} \leq& C_Q 2^{-2 \mathrm{min} \parent{\ell_2,\ell_3}}.
\end{align}

\begin{align}
\expp{\absval{\mathcal G\parent{\bar{\bar g}} - \mathcal G \parent{g}}^{2 \kappa}}
\end{align}
using Theorem 3.3 of \cite{bjork2013monte}.

On the other hand, theorem 3.2 and equation (3.54) states that
\begin{align}
\expp{ \absval{ \mathcal G \parent{g} - \mathcal G \parent{\bar{\bar g}} }^{2 \kappa}  } ^{\frac{1}{2 \kappa}} \leq C_{5}^{CV} \parent{ \parent{\Delta t}^{\frac{1}{2}} + \Delta \tau }.
\end{align}

The weak rate for the discretisation error can be bounded by eq. (4.6):
\begin{align}
\expp{ \absval{ \mathcal G \parent{g} - \mathcal G \parent{\bar{\bar g}} }^{}  }
= 
 \bigo{\parent{\Delta t}^2 + \parent{\Delta \tau}^2}
 = \bigo{2^{-2 \ell_1} + 2^{-2 \mathrm {min} \parent{\ell_2,\ell_3} }}
\end{align}

Due to the fact that the discretisation errors relating to $\tau$
depend on the minimum of $\ell_2$ and $\ell_3$, it seems natural
to assume $\ell_3=\ell_2$. This gives us a two-dimensional
Multi-Index Monte Carlo formulation with the following effective parameters
\begin{align}
\bar \beta &= [2,2]^T ,\\
\bar \gamma &= [1,1]^T , \\
\bar s &= [1,2], \\
\bar w & = [1,2].
\end{align}

On the other hand, one may set $\ell_1=\ell_2=\ell_3=\ell$
and arrive at a one-dimensional MLMC formulation with a weak rate of $2$, strong rate of $1$, and computational cost per realisation of $W_\ell=2^{2\ell}$. In the sequel, we will present the results of these two variations for comparison.

\section{Numerical tests}

For the simplest possible example, let our model be Gaussian with
the following exponential covariance structure
\begin{align}
\expp{df \parent{t,\tau} df \parent{t, \tau'}}
=
\sigma^2 \expf{- \kappa \absval{\tau-\tau'}} dt.
\end{align}

Let our first test case be arguably simplest infinite dimensional one possible,
\begin{align*}
F \parent{x} &= 1-x,\\
U \parent{x} &= 0, \\
G \parent{x} &= x, \\
\Psi \parent{x} &= x, \\
f \parent{0,\tau} &= 0, \\
\kappa &= 2.
\end{align*}

In \ref{fig:empInfDimRates} we get see that we have rate one for both 
the weak and the strong rate along both dimensions for this text example.

\begin{figure}
\includegraphics[scale=0.7]{figure_1.pdf}
\caption{Sample realisation of a forward curve with the discretisation level 
$\overline \ell = [5,5,5]^T$.}
\end{figure}

It remains to deduce the cost of simulating the cost of a realisation. In order to
simulate Gaussian random variables one needs the Cholesky decomposition of the covariance matrix. The cost to generate this $N_\tau \times N_\tau$-matrix is $N_\tau^3$, with $N_\tau = 2^{\ell_3} 2^{\ell_2}$. Using the Cholesky decomposition, the cost to take a a time step becomes $N_\tau^2$. Putting all together, the cost to generate $M$ Monte Carlo realisations with $N_t$ time steps and $N_\tau$ discretisation steps along the maturity direction becomes
\begin{align*}
C \parent{\Delta t, \Delta \tau, M} = N_\tau^3 + M N_\tau^2 N_t.
\end{align*}

\begin{figure}
    \centering
    \begin{subfigure}[b]{0.4\textwidth}
        \includegraphics[width=\textwidth]{dim_1_strong.pdf}
        %\caption{A gull}
        \label{fig:gull}
    \end{subfigure}
    ~ %add desired spacing between images, e. g. ~, \quad, \qquad, \hfill etc. 
      %(or a blank line to force the subfigure onto a new line)
    \begin{subfigure}[b]{0.4\textwidth}
        \includegraphics[width=\textwidth]{dim_1_weak.pdf}
        %\caption{A tiger}
        \label{fig:tiger}
    \end{subfigure}
    \\
    ~ %add desired spacing between images, e. g. ~, \quad, \qquad, \hfill etc. 
    %(or a blank line to force the subfigure onto a new line)
    \begin{subfigure}[b]{0.4\textwidth}
        \includegraphics[width=\textwidth]{dim_2_strong.pdf}
        %\caption{A mouse}
        \label{fig:mouse}
    \end{subfigure}
        \begin{subfigure}[b]{0.4\textwidth}
        \includegraphics[width=\textwidth]{dim_2_weak.pdf}
        %\caption{A mouse}
        \label{fig:mouse}
    \end{subfigure}
    \caption{\label{fig:empInfDimRates}Empirical strong and weak convergence rate for the $\Delta t$ and $\Delta \tau$ discretisation errors.}
\end{figure}



\bibliographystyle{agsm}
\bibliography{references}

\appendix




\end{document}


