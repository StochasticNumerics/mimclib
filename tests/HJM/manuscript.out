\BOOKMARK [1][-]{section.1}{1. Intro}{}% 1
\BOOKMARK [1][-]{section.2}{2. Numerical tests}{}% 2
